src/Tests/dotnet.Tests/dotnet-new/DotnetModifyJsonPostActionTests.cs

```diff
@@ -0,0 +1,306 @@
+
+            DotnetModifyJsonPostActionProcessor processor = new DotnetModifyJsonPostActionProcessor();
+
+            bool result = processor.Process(
+                _engineEnvironmentSettings,
+                postAction,
+                new MockCreationEffects(),
+                new MockCreationResult(),
+                targetBasePath);
+
+            Assert.True(result);
+
+            JsonNode modifiedJsonContent = JsonNode.Parse(_engineEnvironmentSettings.Host.FileSystem.ReadAllText(jsonFilePath));
+
+            testCase.AssertionCallback(modifiedJsonContent);
+        }
+
+        private string CreateJsonFile(string targetBasePath, string fileName, string jsonContent)
+        {
+            string jsonFileFullPath = Path.Combine(targetBasePath, fileName);
+            _engineEnvironmentSettings.Host.FileSystem.WriteAllText(jsonFileFullPath, jsonContent);
+
+            return jsonFileFullPath;
+        }
+    }
+
+    public record ModifyJsonPostActionTestCase<TState>(
+        string TestCaseDescription,
+        string OriginalJsonContent,
+        Dictionary<string, string> PostActionArgs,
+        Action<TState> AssertionCallback)
+    {
+        private static readonly ModifyJsonPostActionTestCase<JsonNode>[] s_successTestCases =
+        {
+            new("Can add simple property",
+                @"{""person"":{""name"":""bob""}}",
+                new Dictionary<string, string>
+                {
+                    ["jsonFileName"] = "jsonfile.json",
+                    ["parentPropertyPath"] = "person",
+                    ["newJsonPropertyName"] = "lastName",
+                    ["newJsonPropertyValue"] = "Watson"
+                },
+                (JsonNode modifiedJsonContent) =>
+                {
+                    Assert.NotNull(modifiedJsonContent["person"]["lastName"]);
+                    Assert.Equal("Watson", modifiedJsonContent["person"]["lastName"].ToString());
+                }),
+
+            new("Can add complex property",
+                @"{""person"":{""name"":""bob""}}",
+                new Dictionary<string, string>
+                {
+                    ["jsonFileName"] = "jsonfile.json",
+                    ["parentPropertyPath"] = "person",
+                    ["newJsonPropertyName"] = "address",
+                    ["newJsonPropertyValue"] = @"{""street"": ""street name"", ""zip"": ""zipcode""}"
+                },
+                (JsonNode modifiedJsonContent) =>
+                {
+                    Assert.NotNull(modifiedJsonContent["person"]["address"]);
+                    Assert.Equal("street name", modifiedJsonContent["person"]["address"]["street"].ToString());
+                }),
+
+            new("Can add property to document root",
+                @"{""firstProperty"": ""foo""}",
+                new Dictionary<string, string>
+                {
+                    ["jsonFileName"] = "jsonfile.json",
+                    ["parentPropertyPath"] = null,
```

Comment:
If this value is assumed null when not specified, I would not include this line on the test, so we can check if the minimum requirements are being met / not met. This relates to my previous comment on the `ProcessInternal` function.

Reply:
Ah, I can do that.  I just thought to make it explicit that this property is not specified.